### PR TITLE
Initializing flash messages and adding url_for() in templates

### PR DESCRIPTION
**Pull request by patrzykat**
1. Initialized flash messages **with** Bootstrap categories on layout.html
2. url_for() allows **url** to be easily changed (above each route).

```python
@app.route("/logout/")
def FUN_logout():
    pass
```

could be changed to 

```python
@app.route("/auth/logout/")
def FUN_logout():
    pass
```

without interfering with any of the templates.

I thought this might be (somewhat) helpful. I tested everything after editing and it all works.

### CODE REVIEW

**Comment by XD-DENG** [comment]
Many thanks @patrzykat for your contribution. You can @ me once you feel this PR is ready for a review.

I created this project quite long time ago, so some contents may not be most updated. I would be happy to hear and accept any ideas or suggestions.

In another PR, you mentioned you are new to open source, that's totally fine. Everyone starts as a "newbie". Let me know if you have any question about anything (this project, or open source community, or Apache projects where I'm a community member), would be happy to share.

**Comment by patrzykat** [comment]
@XD-DENG I appreciate it! 

This is most likely the only project I can contribute to at the moment. Everyone starts somewhere :)

File: templates/layout.html [hunk 1]
```diff
@@ -52,6 +52,17 @@
     </div>
     </nav>
 
+{% with messages = get_flashed_messages(with_categories=true) %}
+    {% if messages %}
+        <!-- note that this is in opposite order, I dont know why -->
+        {% for category, message in messages %}
+            <div class="alert alert-{{ category }}">
+                {{ message }}
+            </div>
```
Comment (XD-DENG) [inline templates/layout.html:61]: I wonder why we are not following the example in doc? https://flask.palletsprojects.com/en/1.1.x/patterns/flashing/#flashing-with-categories . We don't have to handle the `div`s and the class details by ourselves.

In terms of the order, I think your comment line can be removed. It's decided by the underlying data structure, and the users should manage this.

**Review by XD-DENG — CHANGES_REQUESTED** [review]

File: templates/layout.html (hunk 1 above)
Comment (patrzykat) [reply templates/layout.html:60]: I removed the comment. It was unnecessary. 

I started with the code from the Flask documentation. It did not look good on the app so I decided to use the "alert" classes built into Bootstrap.  

Thanks for the feedback!

Comment (XD-DENG) [reply templates/layout.html:60]: Yep, I had a try and I get what you meant. 

Cool. All look good. Thanks again, I'm happy to see people work together to make a project better. And I hope this is a helpful experience to you as well :)

**Review by XD-DENG — APPROVED** [review]

**Pull request merged**